test/src/test/groovy/org/openremote/test/provisioning/UserAndAssetProvisioningTest.groovy

```diff
@@ -833,4 +864,618 @@ class UserAndAssetProvisioningTest extends Specification implements ManagerConta
+        config.put(PersistenceService.OR_STORAGE_DIR, tempManagerDir.getAbsolutePath())
+
+
+        and: "the container starts"
+        def container = startContainer(config, defaultServices())
+        def keystoreService = container.getService(KeyStoreServiceImpl.class)
+        def mqttBrokerService = container.getService(MQTTBrokerService.class)
+        def identityService = container.getService(ManagerIdentityService.class)
+        def provisioningService = container.getService(ProvisioningService.class)
+        def mqttHost = "localhost"
+        def mqttPort = 8884
+
+        and: "the client certificate is added to the KeyStoreService"
+        mtlsHelper.addClientCertificateToKeyStoreService(
+            keystoreService,
+            provisionedAccountKeyAlias,
+            keystorePassword,
+            clientKeyPair,
+            clientCert
+        )
+
+        expect: "the keystores should contain the certificates"
+        keystoreService.getKeyStore().containsAlias(provisionedAccountKeyAlias)
+        keystoreService.getTrustStore().containsAlias(provisionedAccountKeyAlias)
+
+        when: "A new service user with the corresponding certificate's fields is created"
+        User serviceUser = new User()
+                .setServiceAccount(true)
+                .setEnabled(true)
+                .setRealm("master")
+                .setUsername(ProvisionedAccountUserName);
+        serviceUser = identityService.getIdentityProvider().createUpdateUser("master", serviceUser, null, true);
+
+        identityService.getIdentityProvider().updateUserClientRoles(
+                Constants.MASTER_REALM,
+                serviceUser.getId(),
+                Constants.KEYCLOAK_CLIENT_ID,
+                Stream.of(ClientRole.READ_ASSETS, ClientRole.WRITE_ASSETS, ClientRole.WRITE_ATTRIBUTES).map(ClientRole::getValue).toArray(String[]::new)
+        );
+
+        then: "the service user should exist"
+        def fetchedUser = identityService.getIdentityProvider().getUserByUsername("master", "$User.SERVICE_ACCOUNT_PREFIX$ProvisionedAccountUserName")
+        assert fetchedUser != null
+
+        when: "an MQTT client is created with mTLS configuration"
+        def device1UniqueId = "device1"
+        def mqttDevice1ClientId = device1UniqueId
+        device1Client = new MQTT_IOClient(
+            mqttDevice1ClientId,
+            mqttHost,
+            mqttPort,
+            true,
+            false,
+            null,
+            null,
+            null,
+            keystoreService.getKeyManagerFactory(provisionedAccountKeyAlias),
+            keystoreService.getTrustManagerFactory()
+        )
+
+        device1Client.connect()
+
+        then: "the client should connect successfully using mTLS"
+        conditions.eventually {
+            assert device1Client.getConnectionStatus() == ConnectionStatus.CONNECTED
+        }
+
+        and: "the authenticated client should have the correct subject"
+        conditions.eventually {
```

Comment:
Integration test should test more behaviour, so they are authenticated and now they should be able to publish attribute events, etc.